www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+
+import { isNil, path } from 'ramda';
+import { useAtomValue } from 'jotai/utils';
+import { useTranslation } from 'react-i18next';
+
+import { Paper, Stack, Typography } from '@mui/material';
+import NotificationIconOff from '@mui/icons-material/NotificationsOff';
+import NotificationIconActive from '@mui/icons-material/NotificationsActive';
+import PersonIcon from '@mui/icons-material/Person';
+import GroupIcon from '@mui/icons-material/Group';
+
+import { getData, useRequest } from '@centreon/ui';
+
+import { detailsAtom } from '../../detailsAtoms';
+import {
+  labelAlias,
+  labelContactGroups,
+  labelContacts,
+  labelEmail,
+  labelName,
+  labelNotificationStatus,
+} from '../../../translatedLabels';
+
+import { Contact, ContactGroup, NotificationContacts } from './models';
+import Contacts from './Contacts';
+import ContactsLoadingSkeleton from './ContactsLoadingSkeleton';
+import ContactCell from './ContactCell';
+
+const Notification = (): JSX.Element => {
+  const { t } = useTranslation();
+
+  const [notificationContacts, setNotificationContacts] =
+    React.useState<NotificationContacts | null>(null);
+
+  const { sendRequest } = useRequest<NotificationContacts>({
+    request: getData,
+  });
+  const details = useAtomValue(detailsAtom);
+
+  const endpoint = path(['links', 'endpoints', 'notification_policy'], details);
+
+  const loadNotificationContacts = (): void => {
+    sendRequest({ endpoint }).then(setNotificationContacts);
+  };
+
+  React.useEffect(() => {
+    if (isNil(details)) {
+      return;
+    }
+    loadNotificationContacts();
+  }, [details]);
+
+  const contactHeaders = (
+    <>
+      <Typography sx={{ fontWeight: 'bold', paddingLeft: 1 }}>
+        {t(labelName)}
+      </Typography>
+      <Typography sx={{ fontWeight: 'bold' }}>{t(labelAlias)}</Typography>
+    </>
+  );
+
+  const contactWithEmailHeaders = (
+    <>
+      {contactHeaders}
+      <Typography sx={{ fontWeight: 'bold' }}>{t(labelEmail)}</Typography>
+    </>
+  );
+
+  const getContactColumns = ({ name, alias }): JSX.Element => {
+    return (
+      <>
+        <ContactCell paddingLeft={1}>{name}</ContactCell>
+
+        <ContactCell>{alias}</ContactCell>
+      </>
+    );
+  };
+
+  const getContactWithEmailColumns = (contact: Contact): JSX.Element => {
```

Comment:
Extract it or memoize it

Reply:
On a component i understand what i do , declare component and set only props i would memoize, but for a function , i have some difficulties :/